src/libparms/db/parmsdb.cpp

```diff
@@ -3371,4 +3445,77 @@ bool ParmsDb::updateLastClientSelfRestartTime(int64_t lastClientRestartTime) {
     return true;
 }
 
+
+bool ParmsDb::selectValueForKey(const std::string &key, std::string &value, const std::string &defaultValue) {
```

Comment:
This kind of functions usually has a "bool &found" parameter and no defaultValue

Reply:
I think using defaultValue can make the caller code easier to read. For exemple:
```
bool found = false;
std::string value;

if(!selectValueForKey("lasLogUpload", value, found)){
    //handle the error
}

if(!found){
   //hanlde the not found
}
// use of the value
```

become:
```
std::string value;

if(!selectValueForKey("lasLogUpload", value, "01/01/2000_00h00m00s")){
    //handle the error
}
// use of the value
```

or:

```
std::string value;
selectValueForKey("lasLogUpload", value, "01/01/2000_00h00m00s")){
// use of the value
```

But I also understand the benefice of keeping the same mechanism across the function when possible